plugins/node/opentelemetry-instrumentation-pg/src/types.ts

```diff
@@ -17,6 +17,7 @@
 import type * as pgTypes from 'pg';
 import type * as api from '@opentelemetry/api';
 import { InstrumentationConfig } from '@opentelemetry/instrumentation';
+import { PgClientConnectionParams } from './internal-types';
```

Comment:
`types.ts` is a public api file for the package, and is not allowed to export internal types. You can read more about the issues it might cause [here](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/GUIDELINES.md#dependencies)

You can alternatively move the `PgClientConnectionParams` interface from `internal-types.ts` to `types.ts`.

Reply:
Great catch. I ended up renaming and adding a comment above the internal type — to make clear how the internal type is different from the related `ConnectionConfig` type that is now exported from @types/pg — and making a separate public type for the hook info, so that it's not coupled to the shape of the internal type (which really is typing an object in the pg library that isn't necessarily even intended to be part of that library's API).